src/cos_agent.py

```diff
@@ -0,0 +1,10 @@
+#
+# Copyright 2023 Canonical, Ltd.
+#
+from charms.grafana_agent.v0.cos_agent import COSAgentProvider
+
+
+class Provider(COSAgentProvider):
+    @property
+    def _scrape_jobs(self):
+        return self._metrics_endpoints()
```

Comment:
May be dropped after https://github.com/canonical/grafana-agent-k8s-operator/pull/223 is complete